From: alexn@dsv.su.se
Subject: [PATCH] USB: Fix use after free in usb/core/devices.c

[PATCH] USB: Fix use after free in usb/core/devices.c

There's a low probability use-after-free here caused by some switched
pointers, st needs to go back to file->private_data

Found by Coverity tool

Signed-off-by: Alexander Nyberg <alexn@dsv.su.se>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


 drivers/usb/core/devices.c |    1 +
 1 files changed, 1 insertion(+)


diff -Nru a/drivers/usb/core/devices.c b/drivers/usb/core/devices.c
--- a/drivers/usb/core/devices.c	2005-03-08 16:39:31 -08:00
+++ b/drivers/usb/core/devices.c	2005-03-08 16:39:31 -08:00
@@ -608,6 +608,7 @@
 		/* we may have dropped BKL - need to check for having lost the race */
 		if (file->private_data) {
 			kfree(st);
+			st = file->private_data;
 			goto lost_race;
 		}
 
